docs/socket-groups.md

```diff
@@ -153,37 +175,47 @@ protection, the mechanism should quickly detect a link as broken so
 that packets lost on the broken link can be resent over the others,
 but no such mechanism has been provided for balancing group.
 
+Please also keep in mind that there's a rule for all groups - one member
+link established is enough for a group to be connection established. This
+might be not always wanted in case of balancing groups, so your application
+might need to do additional check and monitor how many links and which
+ones are currently active. For example, if you need 3 links to balance
+the load and having only 2 of them would be too little to withstand the
+whole transmission, you should wait with connecting until all 3 links
+are established. Important thing here is also that one of the links might
+be at some moment not possible to be established, or one of the links
+can get broken during transmission.
+
 As there could be various ways as to how to implement balancing
 algorithm, there's a framework provided to implement various methods,
 and two algorithms are currently provided:
 
-1. `plain` (default). This is a simple round-robin - next link selected
-to send the next packet is the oldest used so far.
-
-2. `window`. This algorithm is performing cyclic measurement of the
+1. `fixed`. In this algorithm you specify the share that is given to
+particular link manually through the `weight` parameter. The weight
+value in case of this algorithm in balancing groups defines how much of
+a share is given to this link. Important here is that you can easily
+think of the weight values as a percentage of load burden for particular
+link - however in reality the share of the load is calculated as a
+percentage that particular link's weight comprises among the sum of all
+weight values. Additionally, a value of 0 is special and is translated
+into a weight that would make it equal to an average of all weights.
+Be careful here with the non-established and broken links. For example,
+if you have 3 links with weight 10, 20 and 30, it results in a load
+balance of 16.6%, 33.3% and 50% respectively. However if the second link
+gets broken, there are then 2 links with 10 and 30, which results in
+load balance of 25% and 75% respectively. Keep in mind that it's up
+to the application to keep the minimum links allowed and break the
+group link that is unable to withstand the load.
+
+2. `window` (default). This algorithm is performing cyclic measurement of the
 minimum flight window and this way determines the "cost of sending"
 of a packet over particular link. The link is then "paid" for sending
 a packet appropriate "price", which is collected in the link's "pocket".
 To send the next packet the link with lowest state of the "pocket" is
 selected. The "cost of sending" measurement is being repeated once per
 a time with a distance of 16 packets on each link.
```

Comment:
```suggestion
minimum flight window and this way determines the "cost of sending"
of a packet over a particular link. The link is then "paid" for sending
a packet with an appropriate "price", which is collected in the link's "purse".
The link with lowest amount in its "purse" is selected to send the next packet. 
The "cost of sending" measurement is being repeated once per
a time at an interval of 16 packets on each link.
```

Reply:
Rephrased to be more "readable". My original idea was a little bit too "metaphoric".